pyproject.toml

```diff
@@ -41,10 +41,14 @@ dependencies = [
     "ruamel.yaml.clib>=0.2.0,<=0.2.8",
     "prompt-toolkit>=3.0.24,<3.0.39",
     "distro>=1.5.0,<1.9.0",
-    "awscrt>=0.19.18,<=0.22.0",
+    "awscrt==0.23.1",
```

Comment:
Note: this change is currently causing the dependency tests to fail, because the `requirements-doc-lock.txt` is not updated, which needs to occur manually. However, on my developer laptop (Apple M3 Pro) there is no wheel for `awscrt` on Python 3.8, which is what the lockfiles are generated with. Need to investigate further the appropriate solution for that. Since this PR is targeting the feature branch, it should be safe to merge and will resolve before merging to the `v2` branch.

Reply:
This can be fixed/rebased once #9109 is merged, as I updated the docs requirements lockfile there and tests are passing.